specs/Autofill.md

```diff
@@ -0,0 +1,110 @@
+When the general autofill setting is disabled, no suggestions appear, and no new information is saved. When the general autofill setting is enabled, information is saved, suggestions appear and clicking on one will populate the form fields. 
+
+
+# Examples
+
+## Win32 C++
+```cpp
+void SettingsComponent::TogglePasswordAutofill() {
+    wil::com_ptr<ICoreWebView2Settings> settings;
+    webView->get_Settings(&settings);
+    wil::com_ptr<ICoreWebView2Settings4> settings4 = settings.try_query<ICoreWebView2Settings4>();
+    bool enabled;
+    settings4->get_IsPasswordAutofillEnabled(&enabled);
+    settings4->put_IsPasswordAutofillEnabled(!enabled);
+}
+
+void SettingsComponent::ToggleGeneralAutofill() {
+    wil::com_ptr<ICoreWebView2Settings> settings;
+    webView->get_Settings(&settings);
+    wil::com_ptr<ICoreWebView2Settings4> settings4 = settings.try_query<ICoreWebView2Settings4>();
+    bool enabled;
+    settings4->get_IsGeneralAutofillEnabled(&enabled);
+    settings4->put_IsGeneralAutofillEnabled(!enabled);
+}
+```
+
+## .NET, WinRT
+```c#
+
+// This demonstrates a scenario in which a button triggers TogglePasswordAutofill or 
+// ToggleGeneralAutofill. 
+
+private void TogglePasswordAutofill(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsPasswordAutofillEnabled = !settings.IsPasswordAutofillEnabled;
+}
+
+private void ToggleGeneralAutofill(CoreWebView2 sender, CoreWebView2NavigationStartingEventArgs e)
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsGeneralAutofillEnabled = !settings.IsGeneralAutofillEnabled;
+}
+
+```
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+```IDL
+interface ICoreWebView2Settings4
+
+[uuid(f051013e-4bb3-46b2-b6e4-6ee3fe4f43c2), object, pointer_default(unique)]
+interface ICoreWebView2Settings4 : ICoreWebView2Settings3 {
+  /// IsPasswordAutofillEnabled controls whether autofill for password
+  /// information is enabled. The IsPasswordAutofillEnabled property behaves 
+  /// independently of the IsGeneralAutofillEnabled property. When 
+  /// IsPasswordAutofillEnabled is false, password information is auto-populated,
+  /// suggestions are shown and clicking on one will populate the fields, 
+  /// but no new data is saved and no Save/Update Password prompts are displayed. 
+  /// When IsPasswordAutofillEnabled is true, password information is auto-populated, 
+  /// suggestions are shown and clicking on one will populate the fields, new data 
+  /// is saved, and a Save/Update Password prompt is displayed. 
+  /// The default value is `FALSE`.
+  [propget] HRESULT IsPasswordAutofillEnabled([out, retval] BOOL* isPasswordAutofillEnabled);
```

Comment:
The behavior of IsPasswordAutofillEnabled does not match the behavior of the correspondingly-named IsGeneralAutofillEnabled. Giving it a parallel name is misleading.

| Behavior | IsPasswordAutofillEnabled = false | IsGeneralAutofillEnabled = false |
|-|-|-|
| Fields autofilled | Yes | No |
| Suggestions | Yes | No |
| Auto-populate on accepted suggestion | Yes | No |
| Save/Update prompt | No | N/A |
| Input saved | No | No |

Given that setting IsPasswordAutofillEnabled = false does not actually disable password autofill, I propose that it is the IsPasswordAutofillEnabled which is incorrectly named.

How about IsPasswordSaveEnabled? Because all this does is control saving of the password, not the filling or suggestions.

Reply:
+1, assuming the behavior difference is intentional? Why is there a behavior difference?

Reply:
Oh I didn't realize this difference between the two. Maura, is the documentation correct and it works this way because of the chromium implementation we're working with? If so, Raymond's suggestion makes sense to me. Otherwise, can we fix the docs or update the implementation for them to match?

Reply:
@david-risney Yes the documentation is correct in that the behaviour differs because of chromium implementation.  It makes sense to me to change the property name so that it doesn't mirror IsGeneralAutofillEnabled.  I'm not sure if IsPasswordSaveEnabled is indicative enough that the password property also is associated with autofilling/autopopulating.

Reply:
Actually looking back at this IsPasswordSaveEnabled makes sense to me given it's the dominant property for passwords, probably the best option, thanks for the suggestion!